### PR TITLE
feat!(db): split up `Block` and `Output` collections

### LINKED ISSUE #153: Rename `MessageRecord` to `MessageDocument`

### LINKED ISSUE #207: Consolidate `SyncDocument` and `MilestoneDocument`

### LINKED ISSUE #213: Use `InputsCommitment::LENGTH` instead of hardcoding the size
This will land in Bee soon.

### PR DESCRIPTION
**Pull request by grtlr**
Closes #153.
Closes #207.
Closes #213.

### CODE REVIEW

File: bin/inx-chronicle/src/api/stardust/v2/routes.rs [hunk 1]
```diff
@@ -54,39 +54,41 @@ pub fn routes() -> Router {
...
-        is_solid: rec.metadata.as_ref().map(|d| d.is_solid),
-        referenced_by_milestone_index: rec.metadata.as_ref().map(|d| d.referenced_by_milestone_index),
-        milestone_index: rec.metadata.as_ref().map(|d| d.milestone_index),
-        should_promote: rec.metadata.as_ref().map(|d| d.should_promote),
-        should_reattach: rec.metadata.as_ref().map(|d| d.should_reattach),
-        ledger_inclusion_state: rec.metadata.as_ref().map(|d| d.inclusion_state),
-        conflict_reason: rec.metadata.as_ref().map(|d| d.conflict_reason as u8),
+        block_id: metadata.block_id.to_hex(),
+        parents: metadata.parents.iter().map(|id| id.to_hex()).collect(),
+        is_solid: Some(metadata.is_solid),
```
Comment (Alex6323) [inline bin/inx-chronicle/src/api/stardust/v2/routes.rs:85]: According to the spec `is_solid` is a required field.

File: bin/inx-chronicle/src/collector/stardust_inx/mod.rs [hunk 2]
```diff
@@ -264,42 +259,24 @@ impl HandleEvent<MilestoneState> for Solidifier {
...
-                        match block_rec
-                            .metadata
-                            .map(|metadata| metadata.referenced_by_milestone_index)
-                        {
-                            Some(ms_index) => {
-                                log::trace!("Block {} is referenced by milestone {}", block_id.to_hex(), ms_index);
+                match self.db.get_block_metadata(block_id).await? {
+                    Some(metadata) => {
+                        log::trace!(
+                            "Block {} is referenced by milestone {}",
```
Comment (Alex6323) [inline bin/inx-chronicle/src/collector/stardust_inx/mod.rs:265]:
```suggestion
                            "Block `{}` is referenced by milestone `{}`",
```
I dunno....should we backtick identifiers consistently?

File: src/db/collections/block.rs [hunk 3]
```diff
@@ -109,51 +69,101 @@ pub struct TransactionHistoryResult {
...
 
+fn projection(field: &str) -> Option<FindOneOptions> {
+    Some(FindOneOptions::builder().projection(doc! {field: 1 }).build())
+}
+
+/// Implements the queries for the core API.
 impl MongoDb {
-    /// Get milestone with index.
-    pub async fn get_block(&self, block_id: &BlockId) -> Result<Option<BlockRecord>, Error> {
+    /// Get a [`Block`] by its [`BlockId`].
```
Comment (Alex6323) [inline src/db/collections/block.rs:78]: check for intra-doc-links breakage

File: src/db/collections/block.rs [hunk 4]
```diff
@@ -109,51 +69,101 @@ pub struct TransactionHistoryResult {
...
     }
 
-    /// Get the children of a block.
+    /// Get the raw bytes of a [`Block`] by its [`BlockId`].
+    pub async fn get_block_raw(&self, block_id: &BlockId) -> Result<Option<Vec<u8>>, Error> {
+        let result = self
+            .0
+            .collection::<BlockDocument>(BlockDocument::COLLECTION)
+            .find_one(doc! {"_id": bson::to_bson(block_id)?}, None)
+            .await?;
```
Comment (Alex6323) [inline src/db/collections/block.rs:92]: consider factoring this out into a helper `async fn get_block_document(...)`

File: bin/inx-chronicle/src/collector/stardust_inx/mod.rs (hunk 2 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/collector/stardust_inx/mod.rs:265]: I don't think we should be bothering with this tbh

**Pull request merged**